### PR TITLE
tests - shutdown gracefully to enable webview tests

### PR DESCRIPTION
**Pull request by bpasero**
//cc @deepak1556

### CODE REVIEW

File: src/vs/platform/lifecycle/electron-main/lifecycleMainService.ts [hunk 1]
```diff
@@ -550,9 +551,33 @@ export class LifecycleMainService extends Disposable implements ILifecycleMainSe
...
+		// Note: Electron implements a similar logic here:
+		// https://github.com/electron/electron/blob/fe5318d753637c3903e23fc1ed1b263025887b6a/spec-main/window-helpers.ts#L5
+		//
+		if (this.environmentService.isExtensionDevelopment && !!this.environmentService.extensionTestsLocationURI && this.windowCounter === 1) {
+			await Promise.race([
+				timeout(1000),
+				new Promise(c => {
+					const testWindow = BrowserWindow.getAllWindows()[0];
+					if (testWindow && !testWindow.isDestroyed()) {
+						testWindow.on('closed', () => c());
```
Comment (deepak1556) [inline src/vs/platform/lifecycle/electron-main/lifecycleMainService.ts]: Would also want to check the validity of `testWindow.webContents` and listen to the event accordingly for the reasons mentioned here https://github.com/electron/electron/blob/master/spec-main/window-helpers.ts#L7-L20

**Review by deepak1556 — COMMENTED** [review]
LGTM with minor change, should fix the crash 🤞

**Comment by bpasero** [comment]
I actually think this change does not help: https://github.com/microsoft/vscode/runs/440525717#step:14:456

File: src/vs/platform/lifecycle/electron-main/lifecycleMainService.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/platform/lifecycle/electron-main/lifecycleMainService.ts]: @deepak1556 ok see my latest commit

**Pull request merged**